packages/discord.js-utilities/src/lib/MessagePrompter.ts

```diff
@@ -0,0 +1,402 @@
+
+		const emojiIndex = emojis.findIndex((emoji) => (response?.emoji?.id ?? response?.emoji?.name) === emoji);
+		const number = numbers[emojiIndex];
+
+		// prettier-ignore
+		return explicitReturn ? { ...response, number } : number;
+	}
+
+	/**
+	 * This executes the [[MessagePrompter]] and sends the message if [[IMessagePrompterOptions.type]] equals reaction.
+	 * The handler will wait for one (1) reaction.
+	 * @param channel The channel to use.
+	 * @param authorOrFilter The author to validate.
+	 * @returns A promise that resolves to the reaction object.
+	 */
+	public async runReaction(
+		channel: TextChannel | NewsChannel | DMChannel,
+		authorOrFilter: User | CollectorFilter
+	): Promise<IMessagePrompterExplicitReturn | string | EmojiResolvable> {
+		const { timeout, explicitReturn } = this.options;
+		const reactions = this.options.reactions as IMessagePrompterOptionsReaction;
+
+		if (!reactions?.length) throw new TypeError('There are no reactions provided.');
+
+		const response = await this.collectReactions(channel, authorOrFilter, reactions, timeout);
+
+		return explicitReturn ? response : response.reaction ?? response;
+	}
+
+	/**
+	 * This executes the [[MessagePrompter]] and sends the message if [[IMessagePrompterOptions.type]] equals message.
+	 * The handler will wait for one (1) message.
+	 * @param channel The channel to use.
+	 * @param authorOrFilter The author to validate.
+	 * @returns A promise that resolves to the message object received.
+	 */
+	public async runMessage(
+		channel: TextChannel | NewsChannel | DMChannel,
+		authorOrFilter: User | CollectorFilter
+	): Promise<IMessagePrompterExplicitReturn | Message> {
+		const { timeout, explicitReturn } = this.options;
+		this.appliedMessage = await channel.send(this.message);
+
+		const collector = await channel.awaitMessages(this.createMessagePromptFilter(authorOrFilter), {
+			max: 1,
+			time: timeout,
+			errors: ['time']
+		});
+		const response = collector.first();
+
+		if (!response) {
+			throw new Error('No messages received');
+		}
+
+		return explicitReturn
+			? {
+					response,
+					options: this.options,
+					appliedMessage: this.appliedMessage,
+					message: this.message
+			  }
+			: response;
+	}
+
+	private async collectReactions(
+		channel: TextChannel | NewsChannel | DMChannel,
+		authorOrFilter: User | CollectorFilter,
+		reactions: string[] | EmojiIdentifierResolvable[],
+		timeout: number
+	): Promise<IMessagePrompterExplicitReturn> {
+		this.appliedMessage = await channel.send(this.message);
+
+		const collector = this.appliedMessage.createReactionCollector(this.createReactionPromptFilter(reactions, authorOrFilter), {
+			max: 1,
+			time: timeout
+		});
+
+		let resolved = false;
+		const collected: Promise<MessageReaction> = new Promise((resolve) =>
+			collector.on('collect', (r) => {
+				resolve(r);
+				resolved = true;
+				collector.stop();
+			})
+		);
```

Comment:
What happens if the message, channel, or guild is deleted (all of which emit `end`)? This promise would hang forever.